File: src/main/java/com/annotation/SocketAction.java
package com.annotation;

import java.lang.annotation.*;

/**
 * 数据处理标签
 */
@Documented
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.METHOD, ElementType.ANNOTATION_TYPE})
public @interface SocketAction {

    /** 注册的事件编号 */
    int value();

}

File: src/main/kotlin/com/annotation/SocketAction.kt
package com.annotation

/**
 * 数据处理标签
 */
@MustBeDocumented
@Retention(AnnotationRetention.RUNTIME)
@Target(
    AnnotationTarget.FUNCTION,
    AnnotationTarget.PROPERTY_GETTER,
    AnnotationTarget.PROPERTY_SETTER,
    AnnotationTarget.ANNOTATION_CLASS
)
annotation class SocketAction(
    /** 注册的事件编号  */
    val value: Int
)
